### PR TITLE
Dev docs

### PR DESCRIPTION
**Pull request by komeP**
* Move the development installation information from the main installation guide into a new set of development setup notes.
* Expand the instructions to describe why virtualenvs are needed.
* Use pip instead of setup.py to install from source.
* Clarify that --pre is sufficient for installing the newest code in master without making a repository clone.
* Add instructions for building documentation.
* Developer Tutorials -> Developer Resources
* Users may already have virtualenv or pyvenv.

### CODE REVIEW

File: docs/Users/Tutorials/Development_Setup.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Development Setup Notes
+=======================
+
+The following are some useful notes for setting up an environment to work on
+coala.
+
+Virtualenv
+----------
+
+We highly recommend installing coala in a virtualenv for development.  This
```
Comment (underyx) [inline docs/Users/Tutorials/Development_Setup.rst]: Two spaces 🙀

File: docs/Users/Tutorials/Development_Setup.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+of standard bears.  You can fork and clone the repositories from:
+
+https://github.com/coala-analyzer/coala
+https://github.com/coala-analyzer/coala-bears
+
+Installing from Git
+-------------------
+
+We recommend first installing the latest development snapshot of coala's
+master branch from and all of its dependencies from pip using
```
Comment (underyx) [inline docs/Users/Tutorials/Development_Setup.rst]:
```- from pip
+ with pip/via pip```

**Comment by sils1297** [comment]
@komeP sorry didn't see your PR, can you make a rebase? :/

**Comment by komeP** [comment]
reack
a8cde5b Docs: Clarify that users may have pyvenv
5a05253 Docs: Change Developer Tutorials -> Resources
c3acb62 Docs: Create a set of notes for development setup

Rebased on top of changes that are not affected by documentation changes.

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**